lib/decorator.js

```diff
@@ -53,25 +74,33 @@ Decorator.prototype.concreteAssert = function (invocation, context) {
     var thisObj = invocation.thisObj;
     var args = invocation.values;
     var message = invocation.message;
+    var enhanced = invocation.enhanced;
     if (context && typeof this.config.modifyMessageBeforeAssert === 'function') {
         message = this.config.modifyMessageBeforeAssert({originalMessage: message, powerAssertContext: context});
     }
     args = args.concat(message);
+
+    var data = {
+        originalMessage: message,
+        enhanced: enhanced,
+        args: args
+    };
+
+    if (context) {
+        data.powerAssertContext = context;
+    }
+
     var ret;
     try {
         ret = func.apply(thisObj, args);
     } catch (e) {
-        if (context) {
-            return this.onError({error: e, originalMessage: message, powerAssertContext: context});
-        } else {
-            return this.onError({error: e, originalMessage: message, args: args});
-        }
-    }
-    if (context) {
-        return this.onSuccess({returnValue: ret, originalMessage: message, powerAssertContext: context});
-    } else {
-        return this.onSuccess({returnValue: ret, originalMessage: message, args: args});
+        data.type = 'error';
```

Comment:
`assertionThrew` - and make it a `boolean`

(AVA has some assertions that don't throw, but return promises that might be rejected)